### DISCUSSION TITLE
Caching, RAM vs Disk

### DISCUSSION THREAD

**Discussion by roxaos** [mpv-player/mpv#15076] [Q&A]
Apologies if this is a silly question but, is there any advantage/disadvantage to how/where the cache is stored?

**Comment by Andarwinux** [comment]
The difference is just performance. Disk cache obviously doesn't make sense for local files, but memory cache can still save power on SSD.

**Reply by roxaos** [reply]
I primarily use mpv for streaming video through streamlink. Assuming capacity isn’t an issue would memory still be preferable to disk for storing the cache?

**Reply by Andarwinux** [reply]
Yes, but there are some "design flaws" in the current memory cache, namely the speed of deallocation. On most platforms this isn't really an issue since malloc is fast enough, with the exception of early Windows (pre-2004) where SegmentHeap was not available.

ref:
https://github.com/mpv-player/mpv/pull/14955
https://github.com/mpv-player/mpv/pull/13137

**Reply by roxaos** [reply]
So if I’m understanding this correctly, those design flaws would introduce higher cpu load when encountered, and cache-to-disk doesn’t have such problems?

I’m currently using windows 11.

**Reply by Andarwinux** [reply]
> So if I’m understanding this correctly, those design flaws would introduce higher cpu load when encountered, and cache-to-disk doesn’t have such problems?
> 
> I’m currently using windows 11.

It's just slow, but it has very little impact on CPU load, because NTHeap is very unscalable. This is relative, of course: although the memory cache is a bit “slow”, it's still much faster than the disk cache. For Windows 11, SegmentHeap is always available, so as long as there's enough memory, memory cache is always beneficial.

Of course, if you have something uses LargePages, the memory cache should be avoided, otherwise it will increase memory fragmentation to the point where Windows will have a harder time finding 2M pages.
